src/olympia/lib/git.py

```diff
@@ -525,6 +532,46 @@ def _render_patch(self, patch, commit, parent):
                 'changes': changes
             })
 
+        # We are exposing unchanged files fully to the frontend client
+        # so that it can show them for an better review experience.
+        # We are using the "include unmodified"-flag for git but that
+        # doesn't render any hunks and there's no way to enforce it.
+        # Unfortunately that means we have to simulate line changes and
+        # hunk data for unmodified files.
+        # Unchanged files are *only* exposed in case of explicitly requesting
+        # a diff view for an file. That way we increase performance for
+        # reguar unittests and full-tree diffs.
+        generate_unmodified_fake_diff = (
+            not patch.delta.is_binary and
+            pathspec is not None and
+            patch.delta.status == pygit2.GIT_DELTA_UNMODIFIED
+        )
+        if generate_unmodified_fake_diff:
+            tree = self.get_root_tree(commit)
+
+            blob_or_tree = tree[patch.delta.new_file.path]
+
+            actual_blob = self.git_repository[blob_or_tree.oid]
+
+            changes = [
+                {
+                    'content': line,
+                    'type': GIT_DIFF_LINE_CONTEXT,
+                    'old_line_number': lineno,
+                    'new_line_number': lineno,
+                }
+                for lineno, line in enumerate(actual_blob.data.split(b'\n'))
+            ]
+
+            hunks.append({
+                'header': '@@ -0 +0 @@',
```

Comment:
@bobsilverberg 	I'm going to merge this PR to test it myself on -dev even maybe but we may need to tweak the faked hunk a bit, depending on the parsing of the used library. I looked into the source-code, and the fake should work relatively fine - just so that you're aware in case you run into any kinds of issues while testing this.